cmd/kubebuilder/util/stdin.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bufio"
+	"log"
+	"os"
+	"strings"
+)
+
+// Yesno reads from stdin looking for one of "y", "yes", "n", "no" and returns
+// true for "y" and false for "n"
+func Yesno() bool {
+	reader := bufio.NewReader(os.Stdin)
+	for {
+		switch readstdin(reader) {
+		case "y", "yes":
+			return true
+		case "n", "no":
+			return false
+		}
```

Comment:
Maybe do something when it's an invalid input.

Reply:
Since this is inside a for loop, it won't proceed until `y` or `n` typed.